inmem.go

```diff
@@ -0,0 +1,260 @@
+package clover
+
+import (
+	"sort"
+	"sync"
+)
+
+//iterating over a map doesn't guarantee order of the elements
+//in order to return same results for every unsorted query
+//we also keep insertion id and sort the map elements before querying
+type document struct {
+	insertId int
```

Comment:
I wonder if this is really necessary, since, if the user want to enforce a particuar order on the output, it could use Sort().

Reply:
Because of the way TestSkip and TestLimitAndSkip test compare the result, they fail if i remove the sorting. It looks like badger iterates on values based on insertion order by default so tests does not fail for badge storage engine. I can modify those test and replace the document type i created with original Document and remove the sorting code

Reply:
Thank you, now it is clear. I'll take some time to decide on this :=)